src/matches.ts

```diff
@@ -0,0 +1,58 @@
+import { purry } from './purry';
+
+function _matches(target: any, condition: any) {
+  if (typeof condition === 'function') return condition(target);
+
+  if (typeof condition === 'object') {
+    return Object.entries(condition).every(([key, value]) =>
+      matches(target[key], value)
+    );
```

Comment:
1. Notice that Object.entries is es2017. You can use a regular for loop to make it compatible for more users.
2. You are calling the purried vesion of matches here, but you can simply call the implementation, there's no point going through purrying here..
```suggestion
    for (const key in condition) {
      if (!_matches(target[key], condition[key])) {
        return false;
      }
    }
    return true;
```

Reply:
From the tsconfig of the project, it seems that Node 10, which supports es2018, is deemed low enough a baseline. But sure.